File: Windows8Phone/CSharp/Plexi/Models/StateModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plexi.Models
{
    public class StateModel : ModelBase
    {
        private string state;
        public string State
        {
            get
            {
                return state;
            }

            set
            {
                state = value;
                NotifyPropertyChanged("State");
            }
        }

        public object Response { get; set; }

        public StateModel()
        {

        }

        public StateModel(string state, object response)
        {
            Set(state, response);
        }

        public void Set(string state, object response)
        {
            this.Response = response;
            this.State = state;
        }

        public void Reset()
        {
            this.Response = null;
            this.state = null;
        }
    }
}

File: Android/Java/Plexi/Plexi/src/main/java/com/stremor/plexi/models/StateModel.java
package com.stremor.plexi.models;

/**
 * Created by jeffschifano on 10/28/13.
 */
public class StateModel<T extends Enum<T>> {
    private T state;
    private Object response;

    public StateModel(T state, Object response) {
        set(state, response);
    }

    public void set(T state, Object response) {
        this.state = state;
        this.response = response;
    }

    public void reset() {
        this.state = null;
        this.response = null;
    }

    public T getState() { return state; }
    public void setState(T value) { this.state = value; }

    public Object getResponse() { return this.response; }
    public void setResponse(Object value) { this.response = value; }
}
